app/scripts/controllers/transactions/index.js

```diff
@@ -1471,24 +1471,17 @@ export default class TransactionController extends EventEmitter {
     const signedEthTx = await this.signEthTx(
       unsignedEthTx,
       fromAddress,
-      ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
-      txMeta.custodyStatus ? txMeta : undefined,
-      ///: END:ONLY_INCLUDE_IN
+      this._shouldDisablePublish(txMeta) ? txMeta : undefined,
     );
 
-    ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
-    if (txMeta.custodyStatus) {
-      txMeta.custodyId = signedEthTx.custodian_transactionId;
-      txMeta.custodyStatus = signedEthTx.transactionStatus;
-
+    if (this._shouldDisablePublish(txMeta, signedEthTx)) {
```

Comment:
Would this be cleaner as `afterSign` and we'd use the callback `boolean` return value to only return here if `false`.

This callback should only contain the fenced logic so settings the custody properties in the metadata and the call to the `transactionUpdateController`.

Reply:
Perfect, created a new optional callback with the fenced code presented in the `transaction/index.js`.